internal/credsretriever/refreshing_cache.go

```diff
@@ -52,6 +54,18 @@ type internalClock func() time.Time
 // type assertion
 var _ credentials.CredentialRetriever = &cachedCredentialRetriever{}
 
+var (
+	promCacheNonRecoverableError = promauto.NewCounter(prometheus.CounterOpts{
```

Comment:
i think we can just keep this as `pod_identity_cache_errors` and may be we might want to capture both recoverable and non-recoverable errors.

Reply:
fixed :) thx!